src/cow-react/api/gnosisSafe/index.ts

```diff
@@ -37,21 +37,32 @@ function _getClient(chainId: number, library: Web3Provider): SafeApiKit | null {
   return client
 }
 
-function createSafeServiceClient(chainId: number, library: Web3Provider): SafeApiKit | null {
-  const url = SAFE_TRANSACTION_SERVICE_URL[chainId]
-  if (!url) {
-    return null
-  }
-
+function _createSafeEthAdapter(library: Web3Provider): EthersAdapter {
   const provider = new Web3Provider(library.send.bind(library) as JsonRpcFetchFunc)
 
-  const ethAdapter = new EthersAdapter({
+  return new EthersAdapter({
     ethers,
     signerOrProvider: provider.getSigner(0),
   })
+}
+
+export function createSafeServiceClient(chainId: number, library: Web3Provider): SafeApiKit | null {
+  const url = SAFE_TRANSACTION_SERVICE_URL[chainId]
+  if (!url) {
+    return null
+  }
+
+  const ethAdapter = _createSafeEthAdapter(library)
   return new SafeApiKit({ txServiceUrl: url, ethAdapter })
 }
 
+export async function createSafeSdkInstance(safeAddress: string, library: Web3Provider): Promise<Safe> {
+  const ethAdapter = _createSafeEthAdapter(library)
+
+  // TODO: not sure whether we'll need to prefix the address with the chain short code
```

Comment:
I don't think so, why do you think so? 
I guess is working without it, right? I would delete the TODO if so